model/tensorflow/tests/test_dnnr.py

```diff
@@ -0,0 +1,124 @@
+        cls.model = DNNRegressionModel(
+            DNNRegressionModelConfig(
+                directory=cls.model_dir.name,
+                steps=1000,
+                epochs=30,
+                hidden=[10, 20, 10],
+                label_name="TARGET",
+            )
+        )
+        cls.feature1 = Feature_1()
+        cls.feature2 = Feature_2()
+        cls.features = Features(cls.feature1, cls.feature2)
+        # Generating data f(x1,x2) = 2*x1 + 3*x2
+        _n_data = 1000
+        _temp_data = np.random.rand(2, _n_data)
+        cls.repos = [
+            Repo(
+                "x" + str(random.random()),
+                data={
+                    "features": {
+                        cls.feature1.NAME: float(_temp_data[0][i]),
+                        cls.feature2.NAME: float(_temp_data[1][i]),
+                        "TARGET": 2 * _temp_data[0][i] + 3 * _temp_data[1][i],
+                    }
+                },
+            )
+            for i in range(0, _n_data)
+        ]
+        cls.sources = Sources(
+            MemorySource(MemorySourceConfig(repos=cls.repos))
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.model_dir.cleanup()
+
+    async def test_config(self):
+        # Setting up configuration for model
+        config = self.model.__class__.config(
+            parse_unknown("--model-label_name", "TARGET")
+        )
+        self.assertEqual(
+            config.directory,
+            os.path.join(
+                os.path.expanduser("~"), ".cache", "dffml", "tensorflow"
+            ),
+        )
+        self.assertEqual(config.steps, 3000)
+        self.assertEqual(config.epochs, 30)
+        self.assertEqual(config.hidden, [12, 40, 15])
+        self.assertEqual(config.label_name, "TARGET")
+
+    async def test_00_train(self):
+        async with self.sources as sources, self.features as features, self.model as model:
+            async with sources() as sctx, model(features) as mctx:
+                await mctx.train(sctx)
+
+    async def test_01_accuracy(self):
+        async with self.sources as sources, self.features as features, self.model as model:
+            async with sources() as sctx, model(features) as mctx:
+                res = await mctx.accuracy(sctx)
+                self.assertGreater(res, 0.9)
+
+    async def test_02_predict(self):
+
+        a = Repo(
+            "a",
+            data={
+                "features": {self.feature1.NAME: 1.5, self.feature2.NAME: 2}
+            },
+        )
+        async with Sources(
+            MemorySource(MemorySourceConfig(repos=[a]))
+        ) as sources, self.features as features, self.model as model:
+            async with sources() as sctx, model(features) as mctx:
+                res = [repo async for repo in mctx.predict(sctx.repos())]
+                self.assertEqual(len(res), 1)
+            self.assertEqual(res[0].src_url, a.src_url)
+            self.assertTrue(res[0].prediction().value)
```

Comment:
Rather than `assertTrue`, can we add a check that the error of `res[0].prediction().value` for the function `f(x1,x2) = 2*x1 + 3*x2` is not more than `20%`?

Similar to https://github.com/intel/dffml/blob/8577ca57a0bad2c63b860cdd585ae6ca3e4e8fe9/model/scikit/tests/test_scikit.py#L100-L104 only `correct` will need to be calculated.

Reply:
Added